src/tests/uri_tests.cxx

```diff
@@ -1,138 +1,188 @@
+#include <gmock/gmock-actions.h>
 #include "../uri.h"
-#include <gtest/gtest.h>
 
-using namespace uri;
```

Comment:
What's the reason for removing this `using` and instead preprending `uri::` in front of every declaration? I don't mind much, just wondering.

Reply:
I wasn't sure if we were using this but: 

In the [Google Style Guide](https://google.github.io/styleguide/cppguide.html#Namespaces) it says: 
>You may not use a using-directive to make all names from a namespace available.

`using namespace` brings in everything from that namespace so for example if you had two functions with the same name, one from the `uri` namespace and a fake/stub in the tests, it would collide

I think even if we aren't using it I can see why it could cause issues, so might be best practice to avoid using the `using namespace` directive.

Reply:
Sure, it's a good point especially if the namespace would contain many items.
As alternative, there would be also `using uri::URI` to be more specific.